wasi/wasi_test.go

```diff
@@ -126,7 +129,8 @@ func TestArgsAPISucceed(t *testing.T) {
 				opts = append(opts, argsOpt)
 			}
 			wasiEnv := NewEnvironment(opts...)
-			hostFunctionCallContext := buildHostFunctionCallContext()
+			store, err := instantiateWasmStore("testdata/args.wasm", wasiEnv)
```

Comment:
I ended up changing all WASI API calls with `buildHostFunctionCallContext` to invocation through `wat` rather than adding new tests with the wat module invocation which tests only a part of test cases of the existing tests or duplicated test cases. 
However, I also remember that @codefromthecrypt recommended to add tests instead of removing tests (I guess it includes modifying as well) as follows, let me know if this doesn't align with your original suggestion.
```
rather adding a test that does this. you don’t need to remove the tests you made
```

Reply:
I like this better, just I wanted to give you flexibility to keep the old ones if you wanted